x/accreditations/internal/keeper/keeper.go

```diff
@@ -0,0 +1,192 @@
+	}
+
+	// Build the accreditation
+	accreditation := types.Accreditation{
+		Accrediter: accrediter,
+		User:       user,
+		Rewarded:   false,
+	}
+
+	// Save the accreditation
+	accreditationBz := keeper.cdc.MustMarshalBinaryBare(accreditation)
+	store.Set(user, accreditationBz)
+	return nil
+}
+
+// GetAccreditation allows to get the accrediter a given user
+func (keeper Keeper) GetAccreditation(ctx sdk.Context, user sdk.AccAddress) types.Accreditation {
+	store := ctx.KVStore(keeper.StoreKey)
+
+	var accreditation types.Accreditation
+	keeper.cdc.MustUnmarshalBinaryBare(store.Get(user), &accreditation)
+	return accreditation
+}
+
+// GetAccreditations returns all the accreditations that have been
+func (keeper Keeper) GetAccreditations(ctx sdk.Context) (accreditations []types.Accreditation) {
+	store := ctx.KVStore(keeper.StoreKey)
+	iterator := store.Iterator(nil, nil)
+
+	for ; iterator.Valid() && string(iterator.Key()) != types.TrustedSignersStoreKey; iterator.Next() {
+		var accreditation types.Accreditation
+		keeper.cdc.MustUnmarshalBinaryBare(iterator.Value(), &accreditation)
+		accreditations = append(accreditations, accreditation)
+	}
+
+	return
+}
+
+// DepositIntoPool allows anyone to deposit into the liquidity pool that
+// will be used when giving out rewards for accreditations.
+func (keeper Keeper) DepositIntoPool(ctx sdk.Context, depositor sdk.AccAddress, amount sdk.Coins) error {
+	if amount.IsAnyNegative() {
+		return errors.New("amount cannot be negative")
+	}
+
+	store := ctx.KVStore(keeper.StoreKey)
+
+	// Remove the coins from the user wallet
+	_, err := keeper.BankKeeper.SubtractCoins(ctx, depositor, amount)
+	if err != nil {
+		return err
+	}
+
+	// Add the amount to the pool
+	var pool sdk.Coins
+	keeper.cdc.MustUnmarshalBinaryBare(store.Get([]byte(types.LiquidityPoolKey)), &pool)
+	pool = pool.Add(amount)
+	store.Set([]byte(types.LiquidityPoolKey), keeper.cdc.MustMarshalBinaryBare(&pool))
+
+	return nil
+}
+
+// SetPoolFunds allows to set the current pool funds amount
+func (keeper Keeper) SetPoolFunds(ctx sdk.Context, pool sdk.Coins) {
+	store := ctx.KVStore(keeper.StoreKey)
+	store.Set([]byte(types.LiquidityPoolKey), keeper.cdc.MustMarshalBinaryBare(&pool))
+}
+
+// GetPoolFunds return the current pool funds for the given context
+func (keeper Keeper) GetPoolFunds(ctx sdk.Context) sdk.Coins {
+	store := ctx.KVStore(keeper.StoreKey)
+	var pool sdk.Coins
+	keeper.cdc.MustUnmarshalBinaryBare(store.Get([]byte(types.LiquidityPoolKey)), &pool)
+	return pool
+}
+
+// DistributeReward allows to give the specified accrediter the specified amount of reward related
+// to the accreditation of the specified user
+func (keeper Keeper) DistributeReward(ctx sdk.Context, accrediter sdk.AccAddress, reward sdk.Coins, user sdk.AccAddress) error {
+	if reward == nil {
+		return errors.New("reward cannot be empty")
+	}
+
+	if reward.IsAnyNegative() {
+		return errors.New("rewards cannot be negative")
+	}
+
```

Comment:
I've added the double check so that even if other modules will try to call this method using the keeper directly, we won't later run into negative amount being used.